README.md

```diff
@@ -52,7 +52,7 @@ If you want to **accept** jobs from Faktory, use `Consumer`.
 use faktory::ConsumerBuilder;
 use std::io;
 let mut c = ConsumerBuilder::default();
-c.register("foobar", |job| -> io::Result<()> {
+c.register("foobar", |job: Job| -> io::Result<()> {
```

Comment:
The fact that this annotation is now required everywhere is an indicator that this change as-is would be a breaking change. See the comments later though which I think will allow us to leave this as-is and avoid the breaking change.